### ISSUE TITLE
M1 Mac: Xcode build is missing expected TARGET_BUILD_DIR build setting

### ISSUE THREAD

**Issue by synapsecode** [flutter/flutter#141968]
### Steps to reproduce

I'm not sure if this is reproducible, it just happened to me randomly

### Expected results

I expect the flutter run and flutter build ios commands to work as expected but it fails after taking a really long time.

### Actual results

1. IOS builds take 10+ minutes to get done and then fails saying:
	```
	Xcode build is missing expected TARGET_BUILD_DIR build setting.
	Could not build the application for the simulator.
	```

###  Platform:
```
	macbook air M1 2020
	macos ventura 13.5
	Flutter 3.13.9
	Xcode Version 14.3.1 (14E300c)
```

###  Observations:
1. `flutter build ios` fails with same error
2. Going to XCode, building and then launching in simulator works
3. Connecting real device, worked once and when i terminated the app, i could not launch it next time. same error
4. flutter build ipa --release --export-method ad-hoc works without any issues
5. created a new flutter project, it ran once and then after termination same error came backj

###  Earliest fault i could remember
1. tried to debug in vscode and breakpoints were gray
2. added a debug configuration in vscode
3. app worked once but after termination it stopped working
4. never worked since

###  What i tried
1. flutter clean & flutter pub get & flutter run
2. deleting podfile lock, removing trunk, running podinstall pod update
3. reinstalling same version of flutter 
4. restarting emulator; changing emulator; clearing emulator data
5. restarting mac
6. restarting vscode

Nothing has worked so far. Please suggest a fix, it is very frustrating

### Flutter Doctor output

<details open><summary>Doctor output</summary>

```console
[✓] Flutter (Channel stable, 3.13.9, on macOS 13.5 22G74 darwin-arm64, locale en-GB)
[✓] Android toolchain - develop for Android devices (Android SDK version 33.0.1)
[✓] Xcode - develop for iOS and macOS (Xcode 14.3.1)
[✗] Chrome - develop for the web (Cannot find Chrome executable at
    /Volumes/ExternalSSD/ExternalApps/Google Chrome.app/Contents/MacOS/Google
    Chrome)
    ! /Volumes/ExternalSSD/ExternalApps/Google Chrome.app/Contents/MacOS/Google
      Chrome is not executable.
[✓] Android Studio (version 2022.3)
[✓] VS Code (version 1.85.2)
[✓] Connected device (2 available)
[✓] Network resources

```

</details>

**Comment by darshankawar** [comment]
> but it fails after taking a really long time.

@synapsecode Does it take long time everytime you try to run ? Can you confirm if you have stable / fast network connection   ?
Can you also try to run from xcode by opening xcworkspace and check if you still get same error or not ?


> 5\. it ran once and then after termination same error came back

After this, can you run `flutter doctor -v` and check if network resources section gives any warning ?

**Comment by synapsecode** [comment]
hi, thanks for the quick reply
My network speed is good (86mbps reported by fast.com)

i had to generate an ipa file for a client so i used
```
 flutternext build ipa --release --export-method ad-hoc   
```
worked without any issues

next, as you suggested i opened xcworkspace and selected emulator
and clicked on the play button
App Opened successfully on the emulator

### Flutter Doctor Network Output
```
[✓] Network resources
    • All expected network resources are available.
```

after all this,
1. flutter build ios (WORKS)
2. VSCODE run without debugging => (WORKS) => Terminate Pressed in VsCode
3. again VSCODE run without debugging => (WORKS)
4. Moving to another project
	1. ran pod install
	2. flutter build ios => (WORKS)
	3. deleted existing launch.json file in .vscode folder
	3. VSCODE run without debugging => (WORKS)

In short, everything magically works now
before making the github issue, it didnt work for 2 days continuously

I dont know what changed, this is very weird and frustrating. 
any pointers? I'm ready to close this

**Comment by darshankawar** [comment]
Thanks for the update and trying out different steps. Most probably it was a cache issue which got cleared.
Closing based on above comment.
